spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/core/subscriber/PubSubSubscriberTemplateTests.java

```diff
@@ -302,6 +302,18 @@ public void testSubscribeAndConvert_AndManualNack()
 		assertThat(testListenableFutureCallback.getThrowable()).isNull();
 	}
 
+	@Test
+	public void destroyingBeanClosesSubscriberStub() {
```

Comment:
Great test -- now try verifying that the executor created in `getExecutorProvider.getExecutorProvider()` is shut down. I'd run an integration test with a breakpoint in whatever concrete subclass `ScheduledExecutorService.shutdown()` is returned from `FixedExecutorProvider.create(scheduler.getScheduledExecutor())`, just to see what happens in real life, before trying to make a test.

Note: this might not actually require a unit test, but trying to make one will be an interesting learning experience. So this is the kind of thing to spend < 2 hours on. But verifying that the executor is shut down in real life is important because otherwise we'll create a memory leak.